src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -148,6 +148,25 @@ private async Task CreateBlocksAsync()
                         continue;
                     }
 
+                    // Check if the current version of the node is reaching end of life.
+                    ChainedHeader tip = this.consensusManager.Tip;
+                    int maxSupportedBlockHeightGracePeriod = this.network.Consensus.Options.MaxSupportedBlockHeightGracePeriod;
+                    int maxSupportedMineBlockHeight = this.network.ConsensusOptions.MaxSupportedMineBlockHeight;
+                    if ((maxSupportedBlockHeightGracePeriod > 0) && (tip.Height > (maxSupportedMineBlockHeight - maxSupportedBlockHeightGracePeriod)))
```

Comment:
Should we also add `&& (maxSupportedMineBlockHeight >= tip.Height)` so we don't show the warning when we are past the max height and technically we "reached end life" and error will be thrown by the next if block